docs/rest-api.yml

```diff
@@ -768,6 +778,9 @@ definitions:
   Server:
     type: object
     properties:
+      id:
```

Comment:
Question for reviewers: how are API changes like this communicated? Change log? Release notes? IOW, how does someone know that they can or should see these fields, i.e. maybe I don't see them because JupyterHub I'm talking to is old, or maybe because I'm not an admin and so I can't do `GET /servers`?

Reply:
Should I be updating `docs/source/changelog.md`?

Reply:
Thinking through this a bit more, the root `GET /` route returns the API version so if API changes are documented in the change log properly then I guess people could determine what they should expect that way in client side code though it would be nice if the API reference itself indicated when something was added or changed by version.

As for knowing if `GET /servers` is available that should be pretty easy - if it's not the response will be a `404` so the client needs to fallback to some other behavior.

Reply:
Yes please to updating the changelog. We often forget and then someone has to go through all PRs when they are making the release.

If you look around the code some of the methods/functions where behaviour changed have a note saying when&what changed. It seems like none of the entries in `rest-api.yml` have those so I'd also not add anything for this new route. I think additions (or changes that don't break existing behaviour) are Ok to just announce in the changelog.